--- glib-2.76.3/meson.build	2023-05-23 06:55:59.000000000 -0400
+++ glib-2.76.3-openscad/meson.build	2023-06-16 00:04:05.000000000 -0400
@@ -2038,7 +2040,7 @@
 endif
 
 pcre2_req = '>=10.32'
-pcre2 = dependency('libpcre2-8', version: pcre2_req, required: false, allow_fallback: false)
+pcre2 = dependency('libpcre2-8', version: pcre2_req, required: false, default_options: ['default_library=static'])
 if not pcre2.found()
   if cc.get_id() == 'msvc' or cc.get_id() == 'clang-cl'
   # MSVC: Search for the PCRE2 library by the configuration, which corresponds
