diff --color -ru lib3mf-2.4.1/CMakeLists.txt lib3mf-2.4.1-build/CMakeLists.txt
--- lib3mf-2.4.1/CMakeLists.txt	2025-02-26 10:59:32
+++ lib3mf-2.4.1-build/CMakeLists.txt	2025-10-05 21:32:47
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.0)
+cmake_minimum_required (VERSION 3.10)

 cmake_policy(SET CMP0054 NEW)
 cmake_policy(SET CMP0048 NEW)
@@ -6,8 +6,6 @@

 set_property(GLOBAL PROPERTY USE_FOLDERS ON)

-include(GNUInstallDirs)
-
 # Define Version
 set(LIB3MF_VERSION_MAJOR 2)        # increase on every backward-compatibility breaking change of the API
 set(LIB3MF_VERSION_MINOR 4)        # increase on every backward compatible change of the API
@@ -19,6 +17,8 @@
   DESCRIPTION "An implementation of the 3D Manufacturing Format file standard"
 )

+include(GNUInstallDirs)
+
 find_program(buildcache_program buildcache)
 if(buildcache_program)
   set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${buildcache_program}")
@@ -26,7 +26,7 @@

 set(CMAKE_INSTALL_BINDIR bin CACHE PATH "directory for installing binary files")
 set(CMAKE_INSTALL_LIBDIR lib CACHE PATH "directory for installing library files")
-set(CMAKE_INSTALL_INCLUDEDIR include/lib3mf CACHE PATH "directory for installing header files")
+set(CMAKE_INSTALL_INCLUDEDIR include CACHE PATH "directory for installing header files")

 option(USE_INCLUDED_ZLIB "Use included zlib" ON)
 option(USE_INCLUDED_LIBZIP "Use included libzip" ON)
@@ -201,7 +201,8 @@
 else()
     find_package(PkgConfig REQUIRED)
     pkg_check_modules(LIBZIP REQUIRED libzip)
-    target_link_libraries(${PROJECT_NAME} ${LIBZIP_LIBRARIES})
+    target_link_libraries(${PROJECT_NAME} ${LIBZIP_LINK_LIBRARIES})
+    target_include_directories(${PROJECT_NAME} PRIVATE ${LIBZIP_INCLUDE_DIRS})
 endif()


