diff --git a/opencsg.pro b/opencsg.pro
index b56e622..5cf2d6d 100644
--- a/opencsg.pro
+++ b/opencsg.pro
@@ -1,2 +1,2 @@
 TEMPLATE = subdirs
-SUBDIRS  = src example
+SUBDIRS  = src
diff --git a/src/src.pro b/src/src.pro
index 9091251..598112a 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -7,6 +7,8 @@ TARGET = opencsg
 }
 
 CONFIG += opengl warn_on release
+CONFIG += absolute_library_soname
+QMAKE_STRIP = echo 
 INCLUDEPATH += ../include ../ $$INSTALLDIR/include
 CONFIG -= qt
 LIBS += -L$$INSTALLDIR/lib -lGLEW
