--- a/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h	2021-12-04 14:29:44.000000000 -0500
+++ b/qtbase/src/plugins/platforms/cocoa/qiosurfacegraphicsbuffer.h	2021-12-04 14:29:37.000000000 -0500
@@ -43,6 +43,8 @@
 #include <qpa/qplatformgraphicsbuffer.h>
 #include <private/qcore_mac_p.h>
 
+#include <CoreGraphics/CGColorSpace.h>
+
 QT_BEGIN_NAMESPACE
 
 class QIOSurfaceGraphicsBuffer : public QPlatformGraphicsBuffer
