--- src/3rdparty/libtiff/libtiff/tif_config.h
+++ src/3rdparty/libtiff/libtiff/tif_config.h
@@ -317,15 +317,6 @@
 /* Define to empty if `const' does not conform to ANSI C. */
 /* #undef const */
 
-/* Define to `__inline__' or `__inline' if that's what the C compiler
-   calls it, or to nothing if 'inline' is not supported under any name.  */
-#ifndef Q_OS_SYMBIAN
-#ifndef __cplusplus
-#undef inline
-#define inline
-#endif
-#endif
-
 /* Define to `long int' if <sys/types.h> does not define. */
 /* #undef off_t */
 
