diff -ru qt-everywhere-src-6.5.5/qtbase/cmake/FindWrapOpenGL.cmake qt-everywhere-src-6.5.5-new/qtbase/cmake/FindWrapOpenGL.cmake
--- qt-everywhere-src-6.5.5/qtbase/cmake/FindWrapOpenGL.cmake	2025-05-30 09:52:30
+++ qt-everywhere-src-6.5.5-new/qtbase/cmake/FindWrapOpenGL.cmake	2025-10-08 00:41:01
@@ -37,16 +37,7 @@
             set(__opengl_fw_path "-framework OpenGL")
         endif()
 
-        find_library(WrapOpenGL_AGL NAMES AGL)
-        if(WrapOpenGL_AGL)
-            set(__opengl_agl_fw_path "${WrapOpenGL_AGL}")
-        endif()
-        if(NOT __opengl_agl_fw_path)
-            set(__opengl_agl_fw_path "-framework AGL")
-        endif()
-
         target_link_libraries(WrapOpenGL::WrapOpenGL INTERFACE ${__opengl_fw_path})
-        target_link_libraries(WrapOpenGL::WrapOpenGL INTERFACE ${__opengl_agl_fw_path})
     else()
         target_link_libraries(WrapOpenGL::WrapOpenGL INTERFACE OpenGL::GL)
     endif()
